### PR TITLE
First part Search functionality (+ Station details)

### PR DESCRIPTION
**Pull request by indydedeken**
It's a big functionality, that's why I prefer to split it in multiples parts.
This PR contains: 
- search view: 
  - :ok: with simples markers
  - :ok: allow to search street/city name
  - :ok: access to station details from a markers (with a click), 
- station details view: 
  - :ok: display all informations
- to do - _improvements_:
  - :x: allow to search stations names, 
  - :x: markers with bikes/stands availables,
  - etc...

### CODE REVIEW

File: js/Control.OSMGeocoder.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
+if (typeof console == "undefined") {
+	this.console = { log: function (msg) { /* do nothing since it would otherwise break IE */} };
+}
+
+
+L.Control.OSMGeocoder = L.Control.extend({
```
Comment (eliemichel) [inline js/Control.OSMGeocoder.js:6]: Who is `L`?

Comment (indydedeken) [inline js/Control.OSMGeocoder.js:6]: L represents the Leaflet object.

**Comment by Phyks** [comment]
Sorry for the delay, I have been pretty busy this week.

Two things:
- First of all, I have a sort of infinite loop (according to the console output) when browsing any view, now.
- Concerning the code itself, would it be possible to move [this](https://github.com/indydedeken/VelibFxos/tree/design/leaflet-control-osm-geocoder-master) to a Git submodule, which would be cleaner ?

**Comment by indydedeken** [comment]
@Phyks no problem, about these two things : 
- the inifinite loop is because of an unauthorized access to geolocation. It's linked to the #14.
- I forgot to delete this folder, done in e73d9ca.

**Comment by Phyks** [comment]
The map in the search view should be zoomable, I think.

Apart from this, I'm ok with this PR and merged it. Thanks a lot !

**Pull request merged**